meson.build

```diff
@@ -296,6 +296,51 @@ You can disable X11 support with -Denable-x11=false.''')
     )
 endif
 
+# libxkbregistry
+if get_option('enable-xkbregistry')
+    dep_libxml = dependency('libxml-2.0')
```

Comment:
libxml2 has support for `XInclude` directly available, so it's just a matter of defining the callbacks. expat requires [manual handling of the tags](https://github.com/libexpat/libexpat/blob/master/expat/doc/reference.html#L751).

To keep the code simple and the feature flexible, I went with `XInclude`. For example, my current test config file looks like this:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE xkbConfigRegistry SYSTEM "xkb.dtd">
<xkbConfigRegistry version="1.1" xmlns:xi="http://www.w3.org/2001/XInclude">
 <layoutList>
   <layout>
     <configItem>
       <name>foo</name>
        <shortDescription>foo</shortDescription>
       <description>A totally foo item</description>
       <languageList>
         <iso639Id>eng</iso639Id>
       </languageList>
     </configItem>
   </layout>
   <xi:include href="xkb://rules/evdev.xml" xpointer="xpointer(xkbConfigRegistry/layoutList/*)" />
 </layoutList>

```

Reply:
sorry, hit enter too early, amended the comment above now

(this comment is just for email benefit, will delete it :)

Reply:
I've played with a few things but had other things take priority last week. The main thing I tried was to validate input through the DTD (so we don't have to check things like the name field manually). That was painful (`XInclude` got in the way) but got it to work eventually (will push after cleaning up). The second thing was that `XInclude` is a dead end anyway.

I *suspect* that the two majority use-cases will be adding an option and adding a variant. The latter cannot be handled with pure `XInclude` anyway because you can't  easily merge a new variant *into* an existing system layout. So some other merge strategy becomes inevitable sooner than later and then we just parse two separate XML files. I'm not 100% sure on the exact behaviour and whether it'll be an explicit `<merge>` tag or implied behaviour.

So in summary: `XInclude` will go away, and libxml vs expat can be revisited then too.